hs628_tools/Test.py

```diff
@@ -0,0 +1,133 @@
+
+import numpy as np
+from sklearn.model_selection import train_test_split
+
+
+
+
+
+
+class data_split:
+    """
+    Create class data_split to predict a perfect train and test data split ratio.
+
+    defines method to accomplish this task.
+
+    returns a split ratio.
+
+    Identifies Missing values as Nan for any missing value, return train and test dataframe with uniform division
+    of NAN between train and test data per the data split ratio.
+
+    """
+    def __init__(self, data):
+        self.data = data
+
+
+    def test_train():
+        """
+        Reads in csv file using np.genfromtxt, counts number of NANs in the dataset
+
+        """
+
+        data = np.genfromtxt(csv_datafile, delimiter=',', skip_header=1)
+        np.isnan(data).sum() #counts the number of "NANs" in dat
+        #print data.shape
+
+
+
+        #set calculation standard for specific datalengths, this is based on the general rule of thumb
+        #ratio increases as the length of the data increases, so to ensure we have  enough data to train and test
+        #for the smaller datasets
+        
+        if len(data) <= 500:
+            train = 60
+            test = 40
+        elif len(data) > 500 and row_len <= 10000:
+            train = 70
+            test = 30
+        elif len(data) > 10000 and row_len <=100000:
+            train = 75
+            test = 25
+        else:
+            train = 80
+            test = 20
+        print 'train:',train, 'test:',test
+
+
+        # drops all NAN values from data
+        non_nandata = data[~np.isnan(data).any(axis=1)]
```

Comment:
It is generally not advisable to straight drop all NaN's primarily because you are dropping the entire row, with other column data which might not have a NaN. It should be evaluated based on the percentage of missing values for the column you can decide to impute the data rather than dropping. But anyways this feature doesn't really focus on that aspect so it's okay. But when dealing with small datasets which generally the case with health care data always think of a way to impute without creating bias rather than making them NaN's and dropping the rows.

Reply:
That makes sense. I was very focused in accomplishing the test and train split ratio and may have overlooked the sensitiveness of healthcare data. I will work on getting this included in my codes. Thanks.

-Nirvesh

Reply:
It seems to me that the real issue here is that the comment in the code is so terse that, just looking at this point, it gives the wrong impression that the samples with NaN's are actually being thrown away, whereas really they are simply being dealt with separately.

@nirveshk your response to Rohit's comment makes it look like you yourself forgot what you were doing here, I think you must have rushed your response a bit too much.  But that just emphasizes that the comment in the code should be better.

Reply:
@EHWUSF I agree with your opinion, I may have rushed in replying @rohitchadaram. I was under the impression that the Nan data were actually dropped. However, later I realized this is not the case in my package. I am temporarily dropping rows with NAN values so that I can accumulate all NAN rows in an array and later use them to uniformly distribute(based on the train test ratio) across train and test set.